Update nuki.markdown

AJediIAm opened the pull request:
Add extra suggestions for direct integration.

<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).

  Before submitting your pull request, please verify that you have chosen the correct target branch,
  and that the PR preview looks fine and does not include unrelated changes.
-->
## Proposed change
<!-- 
    Describe the big picture of your changes here to communicate to the
    maintainers why we should accept this pull request. If it fixes a bug
    or resolves a feature request, be sure to link to that issue in the 
    additional information section.
-->



## Type of change
<!--
    What types of changes does your PR introduce to our documentation/website?
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR.
-->

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [x] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logos and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information
<!--
    Details are important, and help maintainers processing your PR.
    Please be sure to fill out additional details, if applicable.
-->

- Link to parent pull request in the codebase: 
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist
<!--
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR. If you're unsure about any of them, don't hesitate to ask.
    We're here to help! This is simply a reminder of what we are going to look
    for before merging your code.
-->

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [ ] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **Documentation**
	- Enhanced Nuki integration docs to clarify control options via Nuki Bridge and WiFi-enabled Smart Locks.
	- Updated prerequisites for connecting devices to WiFi and enabling developer mode.
	- Renamed the MQTT section to "Local integration alternatives" and expanded its content.
	- Added new sections on HomeKit support and Matter support for enhanced compatibility.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

AJediIAm committed:
Update nuki.markdown

Add extra suggestions for direct integration.

AJediIAm committed:
Merge branch 'home-assistant:current' into patch-6

frenck commented on source/_integrations/nuki.markdown:
```diff
@@ -71,7 +71,15 @@ Events generated by Nuki are sent as events of type `nuki_event` with the follow
+
+## Matter Support 
+
+Smart Lock Ultra, Smart Lock Pro, Smart Lock Go, Smart Locks of 4th Generation support Matter over Thread and can directly integrate with Home Assistant.
```
Instead of sprinkling this document with all kind of references to different ways of integrating; I'd suggest to make a single paragraph that explains it all for all of them.

AJediIAm replied:
I'll update it. 
Let me try to figure out how to do this in GitHub. 
Apologies if I hit the wrong button somewhere.

AJediIAm replied:
I made some updates. I hope it shows up correctly...

AJediIAm committed:
Merge branch 'current' into patch-6

AJediIAm committed:
Update nuki.markdown

- Group alternative integrations
- Add references to smart locks with build in wifi
- Add links to Nuki knowledge articles and HA integrations
